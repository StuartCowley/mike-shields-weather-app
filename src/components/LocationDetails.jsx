import React from "react";
import PropTypes from "prop-types";

export default function LocationDetails({ city, country }) {
  return (
    <header className="location-details" data-testid="location-details">
      <h1 className="location-details__title">
        {city}, {country}
      </h1>
    </header>
  );
}

LocationDetails.propTypes = {
  city: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired,
};
