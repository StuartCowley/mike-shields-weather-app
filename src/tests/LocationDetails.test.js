import React from "react";
import { render } from "@testing-library/react";
import LocationDetails from "../components/LocationDetails";
import forecast from "../data/forecast.json";

describe("LocationDetails", () => {
  const { city, country } = forecast.location;

  it("matches snapshot", () => {
    const { asFragment } = render(
      <LocationDetails city={city} country={country} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders city and country", () => {
    const { getByText } = render(
      <LocationDetails city={city} country={country} />
    );
    expect(getByText(`${city}, ${country}`)).toBeInTheDocument();
  });

  it("has has correct css class name", () => {
    const { getByText } = render(
      <LocationDetails city={city} country={country} />
    );
    expect(getByText(`${city}, ${country}`)).toHaveClass(
      "location-details__title"
    );
  });
});
