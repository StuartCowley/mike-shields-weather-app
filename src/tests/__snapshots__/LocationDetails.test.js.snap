// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetails matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="location-details"
    data-testid="location-details"
  >
    <h1
      class="location-details__title"
    >
      Manchester, UK
    </h1>
  </header>
</DocumentFragment>
`;
